ResearchKit/Common/ORKEligibilitySelectionView.m

```diff
@@ -0,0 +1,227 @@
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKEligibilitySelectionView.h"
+#import "ORKHelpers.h"
+#import "ORKAnswerFormat_Internal.h"
+#import "ORKDefines_Private.h"
+
+
+@implementation ORKEligibilityButton
+
++ (id)yesEligibilityButton {
+    return [[ORKEligibilityButton alloc] initWithTitle:ORKLocalizedString(@"BOOL_YES", nil)];
+}
+
++ (id)noEligibilityButton {
+    return [[ORKEligibilityButton alloc] initWithTitle:ORKLocalizedString(@"BOOL_NO", nil)];
+}
+
+- (instancetype)initWithTitle:(NSString *)title {
+    self = [super init];
+    if (self) {
+        [self setTitle:title forState:UIControlStateNormal];
+        [self setTitleColor:[UIColor grayColor] forState:UIControlStateNormal];
+        [self setTitleColor:self.tintColor forState:UIControlStateSelected];
+        self.titleLabel.font = [ORKEligibilityButton defaultFont];
+    }
+    return self;
+}
+
+- (CGSize)intrinsicContentSize {
+    CGRect screenBounds = [[UIScreen mainScreen] bounds];
+    return CGSizeMake(screenBounds.size.width/2 , EligibilityButtonHeight);
+}
+
++ (UIFont *)defaultFont {
+    return [UIFont systemFontOfSize:50.0];
+}
+
+- (void)tintColorDidChange {
+    [super tintColorDidChange];
+    [self setTitleColor:self.tintColor forState:UIControlStateSelected];
+    [self setNeedsDisplay];
+}
+
+@end
+
+
+@implementation ORKEligibilitySelectionView {
+    ORKEligibilityButton *_yesButton;
+    ORKEligibilityButton *_noButton;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    if (self) {
+        // Create the buttons and add them to the view.
+        _yesButton = [ORKEligibilityButton yesEligibilityButton];
+        _noButton = [ORKEligibilityButton noEligibilityButton];
+        
+        [_yesButton addTarget:self
+                       action:@selector(buttonTapped:)
+             forControlEvents:UIControlEventTouchUpInside];
+        [_noButton addTarget:self
+                      action:@selector(buttonTapped:)
+            forControlEvents:UIControlEventTouchUpInside];
+        
+        [self addSubview:_yesButton];
+        [self addSubview:_noButton];
+        
+        // Create a separator for in between the buttons.
+        UIView *separator = [UIView new];
+        separator.backgroundColor = [UIColor lightGrayColor];
+        [self addSubview:separator];
+        
+        // Constraints for the buttons and separator.
```

Comment:
I suggest encapsulating constraint setup on a separate `setUpConstraints` method, for homogeneity with the just merged constraints code refactor.